var/spack/repos/builtin/packages/py-pytest/package.py

```diff
@@ -45,6 +45,7 @@ class PyPytest(PythonPackage):
     # Most Python packages only require setuptools as a build dependency.
     # However, pytest requires setuptools during runtime as well.
     depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-setuptools-scm', type=('build', 'run'))
```

Comment:
From the `setup.py`:
```python
setup_requires=['setuptools-scm'],
```
From [Building and Distributing Packages](https://setuptools.readthedocs.io/en/latest/setuptools.html):

> A string or list of strings specifying what other distributions need to be present in order for the setup script to run.

So it sounds like this might only be a `type='build'` dependency. Can you check if it gets imported anywhere in the installed files?

Reply:
Well spotted. You are right, it does not appear. I also did a full test of it with type='build', and it seems to work fine.